src/dmd/initsem.d

```diff
@@ -374,7 +374,8 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, Type t,
             sc = sc.endCTFE();
         if (i.exp.op == TOK.error)
         {
-            return new ErrorInitializer();
+            assert(cast(ErrorExp) i.exp);
```

Comment:
This `cast` is useless, because `ErrorExp` is an `extern(C++)` class

Reply:
Casting it is even less correct. Again, need to find where i.exp.op was set to TOK.error.